x-pack/plugins/infra/public/components/alerting/logs/expression_editor/criterion.tsx

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useMemo, useCallback } from 'react';
+import {
+  EuiPopoverTitle,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiPopover,
+  EuiSelect,
+  EuiFieldNumber,
+  EuiExpression,
+  EuiFieldText,
+  EuiButtonIcon,
+  EuiFormRow,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { IFieldType } from 'src/plugins/data/public';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { IErrorObject } from '../../../../../../triggers_actions_ui/public/types';
+import {
+  Comparator,
+  Criterion as CriterionType,
+  ComparatorToi18nMap,
+} from '../../../../../common/alerting/logs/types';
+
+const firstCriterionFieldPrefix = i18n.translate(
+  'xpack.infra.logs.alertFlyout.firstCriterionFieldPrefix',
+  {
+    defaultMessage: 'with',
+  }
+);
+
+const successiveCriterionFieldPrefix = i18n.translate(
+  'xpack.infra.logs.alertFlyout.successiveCriterionFieldPrefix',
+  {
+    defaultMessage: 'and',
+  }
+);
+
+const criterionFieldTitle = i18n.translate('xpack.infra.logs.alertFlyout.criterionFieldTitle', {
+  defaultMessage: 'Field',
+});
+
+const criterionComparatorValueTitle = i18n.translate(
+  'xpack.infra.logs.alertFlyout.criterionComparatorValueTitle',
+  {
+    defaultMessage: 'Comparison : Value',
+  }
+);
+
+const getCompatibleComparatorsForField = (fieldInfo: IFieldType | undefined) => {
+  if (fieldInfo?.type === 'number') {
+    return [
+      { value: Comparator.GT, text: ComparatorToi18nMap[Comparator.GT] },
+      { value: Comparator.GT_OR_EQ, text: ComparatorToi18nMap[Comparator.GT_OR_EQ] },
+      { value: Comparator.LT, text: ComparatorToi18nMap[Comparator.LT] },
+      { value: Comparator.LT_OR_EQ, text: ComparatorToi18nMap[Comparator.LT_OR_EQ] },
+      { value: Comparator.EQ, text: ComparatorToi18nMap[Comparator.EQ] },
+      { value: Comparator.NOT_EQ, text: ComparatorToi18nMap[Comparator.NOT_EQ] },
+    ];
+  } else if (fieldInfo?.aggregatable) {
+    return [
+      { value: Comparator.EQ, text: ComparatorToi18nMap[Comparator.EQ] },
```

Comment:
I understand we're using the same comparators here, but using `EQUALS` for both numbers and string values feels confusing. When I switch from this:
<img width="565" alt="Screen Shot 2020-04-23 at 12 03 11 PM" src="https://user-images.githubusercontent.com/1445834/80128115-d5476c00-855a-11ea-89a4-e084322509ba.png">

to this:
<img width="612" alt="Screen Shot 2020-04-23 at 12 03 40 PM" src="https://user-images.githubusercontent.com/1445834/80128136-d9738980-855a-11ea-9b55-dfb13b1f9249.png">

it's not completely clear that the field I've chosen is a different data type.

Maybe change the display message from `EQUALS`/`DOES NOT EQUAL` to `IS`/`IS NOT` for aggregatable fields? That way the word "equals" is only used for numbers.

Reply:
Yeah, I think that's a fair comment, I'll change this.